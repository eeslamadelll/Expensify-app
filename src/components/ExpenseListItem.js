import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, amount, createdAt}) => {
    return (
            <Link className="list-items" to={`/edit/${id}`}>
                <div className="item">
                    <p className="description">{ description }</p>
                    <span className="date">{ moment(createdAt).format('MMMM Do, YYYY') }</span>
                </div>
                <div className="item">
                    <p className="amount">{ numeral(amount / 100).format('$0,0.00') }</p>
                </div>
            </Link>
    )
};

export default ExpenseListItem;