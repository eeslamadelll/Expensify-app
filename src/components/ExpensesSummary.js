import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
    return(
        <div className="page-header">
            <div className="content-container">
                <p>you have <span>{expenseCount}</span> {expenseWord} with total of <span>{numeral(expenseTotal / 100).format('$0,0.00')}</span></p>
                <Link className="add" to="/add">Add Expense</Link>
            </div>
        </div>
    );
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);