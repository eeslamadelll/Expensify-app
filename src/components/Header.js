import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => {
    return (
        <header>
            <h1>Expensify</h1>
            <nav>
                <ul>
                    <li><NavLink to="/dashboard" activeClassName="is-active">Homepage</NavLink></li>
                    <li><NavLink to="/add" activeClassName="is-active">Add Expense</NavLink></li>
                </ul>
                <button onClick={props.startLogout}>Logout</button>
            </nav>
        </header>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(undefined, mapDispatchToProps)(Header);