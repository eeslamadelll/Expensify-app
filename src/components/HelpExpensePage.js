import React from 'react';

const HelpExpensePage = () => {
    return (
        <div>
            Help
        </div>
    );
}

export default HelpExpensePage;