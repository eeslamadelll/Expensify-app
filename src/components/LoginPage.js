import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = (props) => {
    return (
        <div className="login">
            <div className="overlay"></div>
            <div className="container">
                <h1>Expensify App</h1>
                <p>It's time to put your expenses under control</p>
                <button onClick={props.startLogin}>Login with Google</button>
            </div>
        </div>
    );
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => dispatch(startLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage);