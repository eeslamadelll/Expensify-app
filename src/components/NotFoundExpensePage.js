import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundExpensePage = () => {
    return (
        <div>
            404 Not Found Page! <Link to="/">Back Home</Link>
        </div>
    );
}

export default NotFoundExpensePage;