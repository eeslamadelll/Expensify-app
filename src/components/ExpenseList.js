import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container filtered">
		<div className="list-header">
			<div className="show-for-mobile">Expenses</div>
			<div className="show-for-desktop">Expense</div>
			<div className="show-for-desktop">Amount</div>
		</div>
    	{
    		props.expenses.length === 0 ? (
    			<div className="list-items list-items-empty"><p>There is no expenses to show</p></div>
    		) : (
    			props.expenses.map((expense) => {
			        return <ExpenseListItem key={expense.id} {...expense}/>;
			    })
    		)
    	}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);


///////////////////// this code is right but we will use the best practice
// const ConnectExpenseList = connect((state) => {
//     return {
//         expenses: state.expenses,
//         filters: state.filters
//     };
// })(ExpenseList);

// export default ConnectExpenseList;