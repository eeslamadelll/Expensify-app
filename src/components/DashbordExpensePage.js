import React from 'react';
import ExpenseList from './ExpenseList'; // data comes from HOC
import ExpenseListFilters from './ExpenseListFilters';

const DashbordExpensePage = () => {
    return (
        <div>
            <ExpenseListFilters />
            <ExpenseList />
        </div>
    );
}

export default DashbordExpensePage;