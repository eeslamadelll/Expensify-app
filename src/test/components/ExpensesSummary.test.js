import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render the ExpenseSummary correct with one item', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={252} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render the ExpenseSummary correct with multiple items', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={43} expenseTotal={20524752} />);
    expect(wrapper).toMatchSnapshot();
});