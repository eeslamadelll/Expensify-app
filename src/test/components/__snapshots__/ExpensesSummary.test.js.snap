// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the ExpenseSummary correct with multiple items 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <p>
      you have 
      <span>
        43
      </span>
       
      expenses
       with total of 
      <span>
        $205,247.52
      </span>
    </p>
    <Link
      className="add"
      replace={false}
      to="/add"
    >
      Add Expense
    </Link>
  </div>
</div>
`;

exports[`Should render the ExpenseSummary correct with one item 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <p>
      you have 
      <span>
        1
      </span>
       
      expense
       with total of 
      <span>
        $2.52
      </span>
    </p>
    <Link
      className="add"
      replace={false}
      to="/add"
    >
      Add Expense
    </Link>
  </div>
</div>
`;
